Microservices/Posts/src/main/java/cs4337/group_8/TemplateProject/controllers/PostController.java

```diff
@@ -18,14 +18,17 @@ public class PostController {
 
     @Autowired
     private PostService postService;
+    private final JwtService jwtService;
+
 
     @Autowired
     private PostMapper postMapper;
 
     @PostMapping
     public PostDTO createPost(@RequestBody PostDTO postDTO) {
         Post post = postMapper.toEntity(postDTO);
-        return postMapper.toDTO(postService.createPost(post));
+        return postMapper.toDTO(postService.createPost(post, jwtService.generateRefreshToken));
```

Comment:
Why are you making refresh token? That is the Authentication's responsibility. I think you are trying to pass the token from the header. If so add this:
```
    @PostMapping
    public PostDTO createPost(
            @RequestBody PostDTO postDTO,
            @RequestHeader(name="Authorization") String jwtToken
    ) {
        Post post = postMapper.toEntity(postDTO);
        return postMapper.toDTO(postService.createPost(post, jwtToken));
    }

```